subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AttributeBasedFileVisitDetailsFactory.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.file;
+
+import org.gradle.api.file.FileVisitDetails;
+import org.gradle.api.file.RelativePath;
+import org.gradle.internal.nativeintegration.filesystem.FileSystem;
+import org.gradle.internal.os.OperatingSystem;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class AttributeBasedFileVisitDetailsFactory {
```

Comment:
:x: This class and the methods could use some Javadocs.

We seem to have added quite a bit of code in this commit, without adding functionality. 🤔 I'm wondering if we need all of the methods, and if we need all of them to be public.

Reply:
These methods are just extracted from the existing code with some deduplication. 
Three of them are used. The fourth one is added for completeness, I see no harm in it, given that it's only two lines.

I agree with adding Javadoc, though. I'll do it later.